mat64/list_test.go

```diff
@@ -119,12 +125,31 @@ func returnAs(a, t Matrix) Matrix {
 	}
 }
 
+// retranspose returns the matrix m inside an Untransposer of the type
+// of a.
+func retranspose(a, m Matrix) Matrix {
+	_, ok := a.(Untransposer)
```

Comment:
This can be done more elegantly.

```
switch a.(type) {
case Transpose:
    return Transpose{m}
case TransposeTri:
    return TransposeTri{m.(Triangular)}
case Untransposer:
    panic("unknown transpose type")
default:
    panic("not a transposer")
}
```